File: driveappfolderviewer/src/main/java/com/commit451/driveappfolderviewer/FileViewHolder.java
package com.commit451.driveappfolderviewer;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.PopupMenu;
import android.widget.TextView;

import com.google.android.gms.drive.DriveFolder;
import com.google.android.gms.drive.Metadata;

public class FileViewHolder extends RecyclerView.ViewHolder {

    public static FileViewHolder inflate(ViewGroup parent) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_file, parent, false);
        return new FileViewHolder(view);
    }

    TextView mTitleView;
    ImageView mImageView;
    ImageView mMoreView;

    public final PopupMenu popupMenu;

    public FileViewHolder(View view) {
        super(view);
        mTitleView = (TextView) view.findViewById(R.id.file_title);
        mImageView = (ImageView) view.findViewById(R.id.file_image);
        mMoreView = (ImageView) view.findViewById(R.id.file_more);

        popupMenu = new PopupMenu(itemView.getContext(), mMoreView);
        popupMenu.getMenuInflater().inflate(R.menu.menu_file, popupMenu.getMenu());

        mMoreView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                popupMenu.show();
            }
        });
    }

    public void bind(Metadata metadata) {
        if (metadata.getMimeType().equals(DriveFolder.MIME_TYPE)) {
            mImageView.setImageResource(R.drawable.ic_folder_24dp);
        } else {
            mImageView.setImageResource(R.drawable.ic_file_24dp);
        }
        mTitleView.setText(metadata.getTitle());
    }
}

File: driveappfolderviewer/src/main/java/com/commit451/driveappfolderviewer/FileViewHolder.kt
package com.commit451.driveappfolderviewer

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.PopupMenu
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.api.services.drive.model.File

/**
 * View folder for file and folders
 */
internal class FileViewHolder(view: View) : RecyclerView.ViewHolder(view) {

    companion object {

        fun inflate(parent: ViewGroup): FileViewHolder {
            val view = LayoutInflater.from(parent.context)
                    .inflate(R.layout.dafv_item_file, parent, false)
            return FileViewHolder(view)
        }
    }

    private var titleView: TextView = view.findViewById<View>(R.id.file_title) as TextView
    private var imageView: ImageView = view.findViewById<View>(R.id.file_image) as ImageView
    private var moreView: ImageView = view.findViewById<View>(R.id.file_more) as ImageView

    val popupMenu: PopupMenu

    init {
        popupMenu = PopupMenu(itemView.context, moreView)
        popupMenu.menuInflater.inflate(R.menu.dafv_menu_file, popupMenu.menu)

        moreView.setOnClickListener { popupMenu.show() }
    }

    fun bind(file: File) {
        if (file.mimeType == MIME_TYPE_FOLDER) {
            imageView.setImageResource(R.drawable.dafv_ic_folder_24dp)
        } else {
            imageView.setImageResource(R.drawable.dafv_ic_file_24dp)
        }
        var name = file.name
        file.fileExtension?.let {
            name += ".$it"
        }
        titleView.text = name
    }
}